src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,127 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { PeopleFilters } from '../PeopleFilters';
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeopleTable';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+
+export const PeoplePage = () => {
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get('query');
+  const sex = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries');
+  const order = searchParams.get('order');
+  const sort = searchParams.get('sort');
+
+  const [
+    peopleFromServer,
+    setPeopleFromServer,
+  ] = useState<Person[] | null>(null);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    getPeople()
+      .then(setPeopleFromServer)
+      .catch(() => setIsError(true));
+  }, []);
+
+  const filteredPeople = useMemo(() => {
```

Comment:
Your function tooks 40 lines of code in component. Better to move it in separate file, for ex.: `utils/filterPeople.ts`.